neal/sampler.py

```diff
@@ -192,6 +216,9 @@ def sample(self, bqm, beta_range=None, num_reads=10, sweeps=1000,
 
         """
 
+        if not isinstance(bqm, dimod.BinaryQuadraticModel):
+            raise TypeError("'bqm' should be a 'dimod.BinaryQuadraticModel' instance")
```

Comment:
User might want to provide other objects that match the API (though unlikely) or we may want to use other BQM-like classes (for instance if https://github.com/dwavesystems/dimod/pull/477 is merged)

Reply:
Good point. Do we have a way in dimod to check for BQM-like object interface?

Reply:
Not at the moment because we don't yet know what constitutes the 'minimum' bqm. I can generate an issue to make one?

Reply:
Probably creating a BQM super class would make sense here.